java/vector/src/main/java/org/apache/arrow/vector/file/json/JsonFileReader.java

```diff
@@ -332,9 +334,10 @@ private void setValueFromParser(ValueVector valueVector, int i) throws IOExcepti
         ((Float8Vector) valueVector).getMutator().set(i, parser.readValueAs(Double.class));
         break;
       case DECIMAL: {
-          DecimalVector decimalVector = ((DecimalVector) valueVector);
-          byte[] value = decodeHexSafe(parser.readValueAs(String.class));
-          DecimalUtility.writeByteArrayToArrowBuf(value, decimalVector.getBuffer(), i);
+          DecimalVector decimalVector = (DecimalVector) valueVector;
+          BigInteger integerValue = new BigInteger(parser.readValueAs(String.class));
+          BigDecimal decimalValue = new BigDecimal(integerValue, 0);
```

Comment:
Could you add a note that the scale won't be used when writing to the ArrowBuf?  Also, I think you could just use `BigDecimal(integerValue)` to make an unscaled value

Reply:
Yep will do.